server/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -316,11 +317,11 @@ public static QueryBuilder parseInnerQueryBuilder(XContentParser parser) throws
         QueryBuilder result;
         try {
             result = parser.namedObject(QueryBuilder.class, queryName, null);
-        } catch (UnknownNamedObjectException e) {
+        } catch (XContentParseException e) {
```

Comment:
If we're going to use this `XContentParseException` for anything other that unknown named objects we shouldn't catch it here. We need to be sure this is an unknown named object. Either we should rename the exception or make a subclass of it for unknown named objects, I think.

Reply:
Yeah that's probably a good idea, for now there are no other uses of this exception, but if that changes in the future I will make a subclass of it for this.